cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -98,6 +98,54 @@ def counts_to_results(
     return result
 
 
+def _validate_cirq_circuit(circuits: Any) -> None:
+    """Validates that the input is either a single `cirq.Circuit` or a list of `cirq.Circuit`
+    instances.
+
+    Args:
+        circuits: The circuit to run.
+
+    Raises:
+        ValueError: If the input is not a `cirq.Circuit` or a list of
+        `cirq.Circuit` instances.
+    """
+
+    if not (
+        isinstance(circuits, cirq.Circuit)
+        or (
+            isinstance(circuits, Sequence)
+            and all(isinstance(circuit, cirq.Circuit) for circuit in circuits)
+        )
+    ):
+        raise ValueError(
+            "Invalid 'circuits' input. Must be a `cirq.Circuit` or a "
+            "sequence of `cirq.Circuit` instances."
+        )
+
+
+def _validate_get_counts(repetitions: Any) -> None:
+    """Validates that the number of repetitions is positive and
+    an integer.
+
+    Args:
+        repetitions: The number of times to run the circuit.
+
+    Raises:
+        TypeError: If input is not an integer.
+        ValueError: If input is negative.
+    """
+
+    if not (
+        (isinstance(repetitions, int))
+        or (isinstance(repetitions, float) and repetitions.is_integer())
```

Comment:
this will fail for e.g. `repetitions = np.int_(123)`. i think this works though:
```suggestion
        (isinstance(repetitions, SupportsInt) and int(repetitions) == repetitions)
```
(where `SupportsInt` is imported from typing)

Reply:
Ran into coverage issues that protocols could not be used with `isinstance()` and `SupportsInt` so opted for `hasattr()` instead in 5d342e0